app/models/notification_type.rb

```diff
@@ -0,0 +1,38 @@
+class NotificationType < ApplicationRecord
+  AUDIENCE_USER = 'user'.freeze
+  AUDIENCE_TENANT = 'tenant'.freeze
+  AUDIENCE_GLOBAL = 'global'.freeze
+  has_many :notifications
+  validates :message, :presence => true
+  validates :level, :inclusion => { :in => %w(success error warning info) }
+  validates :audience, :inclusion => { :in => [AUDIENCE_USER, AUDIENCE_TENANT, AUDIENCE_GLOBAL] }
+
+  def subscriber_ids(subject, initiator)
```

Comment:
We cannot use `CollectionProxy` just yet. A little hint is the existence of method's attributes.

This pr does not bring `NotificationSubscription` table. So, we do not yet offer a way to subscribe/unsubscribe from the event. Hence we cannot use `CollectionProxy` just yet.

We only return list of users that are by default subscribed as defined by `audience`. Think of `audience` as of strategy for subscribing users. We don't know for sure what strategies we will need. Once we add many notification types, we will have better view into what strategies do we need. So, I suppose we may touch the body of this function in near future.

For example, there could strategy to subscribe users of the given `miq_group`, however we don't know yet about any notification that could potentially fit with this strategy.

It may turn out that we will end-up `NotificationSubsribtion` table full of records, then and only then we can think of `CollectionProxy`